todoApp/spec/main.tsp

```diff
@@ -275,9 +278,18 @@ namespace TodoItems {
 
     @sharedRoute
     @post
-    op createAttachment(
+    op createUrlAttachment(
+      @header contentType: "application/json",
+      @path itemId: TodoItem.id,
+      @body contents: TodoUrlAttachment,
+    ): WithStandardErrors<NoContentResponse | NotFoundResponse>;
+
+   @sharedRoute
+    @post
+    op createFileAttachment(
+      @header contentType: "multipart/form-data",
       @path itemId: TodoItem.id,
-      @body contents: TodoAttachment,
-    ): WithStandardErrors<NoContentResponse | NotFoundErrorResponse>;
+      @body contents: TodoFileAttachment,
```

Comment:
`@multipartBody body: { contents: HttpPart<TodoFileAttachment> }`

Reply:
yeah, I think the original operation has ambiguity where it stands for 1) one part with a file or 2) three parts with contentType, filename or contents?

https://github.com/microsoft/typespec/blob/main/packages/http/lib/http.tsp#L110-L114

Reply:
FileType can be a part of multipart payload but can't be the whole payload type. So current version is not valid case while https://github.com/allenjzhang/typespec-e2e-demo/pull/32/files#r1883276729 is valid.

Reply:
I was wondering if we have any limitation for below case? If no, it would be a valid case standing for a mfd operation which will upload three parts
- string part: contentType
- string part: filename
- bytes(file) part: contents
[playground](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQoNCnVzaW5nIEh0dHA7DQpAcG9zdA0Kb3AgY3JlYXRlRmlsZUF0dGFjaG1lbnQoDQogIEBoZWFkZXIgY29udGVudFR5cGU6ICJtdWx0aXBhcnQvZm9ybS1kYXRhIizFL2JvZHnILXM6IMRSLA0KKTogTm9DxhZSZXNwb25zZSB8IE5vdEZvdW5kyBM7DQo%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)
```
@post
op createFileAttachment(
  @header contentType: "multipart/form-data",
  @body contents: File,
): NoContentResponse | NotFoundResponse;
```

Reply:
> I was wondering if we have any limitation for below case? If no, it would be a valid spec standing for a mfd operation which will upload three parts
> 
> * string part: contentType
> * string part: filename
> * bytes(file) part: contents
> 
> [playground](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQoNCnVzaW5nIEh0dHA7DQpAcG9zdA0Kb3AgY3JlYXRlRmlsZUF0dGFjaG1lbnQoDQogIEBoZWFkZXIgY29udGVudFR5cGU6ICJtdWx0aXBhcnQvZm9ybS1kYXRhIizFL2JvZHnILXM6IMRSLA0KKTogTm9DxhZSZXNwb25zZSB8IE5vdEZvdW5kyBM7DQo%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)
> 
> ```
> @post
> op createFileAttachment(
>   @header contentType: "multipart/form-data",
>   @body contents: File,
> ): NoContentResponse | NotFoundResponse;
> ```

As I said in https://github.com/allenjzhang/typespec-e2e-demo/pull/32/files/052143e392c541fd03a2eaffd7a80f689b499720#r1887849929, I think we should follow @weidongxu-microsoft's comment https://github.com/allenjzhang/typespec-e2e-demo/pull/32/files/052143e392c541fd03a2eaffd7a80f689b499720#r1883276729